### ISSUE TITLE
TypeError: Cannot read property 'block' of undefined     at Function.markdownItContainer

### ISSUE THREAD

**Issue by mesqueeb** [quasarframework/quasar-ui-qmarkdown#52]
For a legitimate bug, in order to process faster, please go here https://codesandbox.io/s/qcalendar-bx1y7, fork the codesandbox, add code to reproduce the bug and submit link here:

**Describe the bug**

> TypeError: Cannot read property 'block' of undefined
    at Function.markdownItContainer (index.esm.js?bc3c:6)
    at MarkdownIt.use (index.esm.js?bc3c:6)
    at extendContainers (index.esm.js?bc3c:6)

**To Reproduce**
1. `git clone https://github.com/mesqueeb/quasar-ui-easy-forms
2. `cd ui && npm i`
3. `npm run dev`
4. See error in console

**Screenshots**
<img width="794" alt="image" src="https://user-images.githubusercontent.com/3253920/69911634-71b08b00-1461-11ea-8518-e985dd42e201.png">

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] macOS
 - Browser [e.g. chrome, safari] chrome
 - Version [e.g. 22] latest

**Comment by mesqueeb** [comment]
Fixed it by doing: `no-container="true"`

not sure what the bug is though.

**Comment by hawkeye64** [comment]
yeah, I have no idea. Will investigate.

**Comment by hawkeye64** [comment]
I'd say, please use `yarn` and see if the issue persists. Don't forget to remove `node_modules` and `package.lock` (or, whatever it's called) file.

**Comment by hawkeye64** [comment]
@mesqueeb By chance do you have two separate instances of QMarkdown being loaded? One being used by one component and another used by a different component?

**Comment by hawkeye64** [comment]
Have a feeling markdownItContainer is doing something globally that conflicts.

**Comment by apasov** [comment]
I also started getting this error after upgrading to v1.0.12
In my case there are two errors. In addition to `Cannot read property 'block' of undefined` I get `Cannot read property 'render' of undefined`:

![image](https://user-images.githubusercontent.com/28765662/71786769-16e9f080-3018-11ea-88ec-1fb094e79014.png)
![image](https://user-images.githubusercontent.com/28765662/71786774-2832fd00-3018-11ea-82b6-d2926d699630.png)

Both are fixable by `:no-container="true"`

**Comment by SouthRibbleTech** [comment]
I am getting the same error

:no-container="true" works but also gives me this error in the console.

`
 Invalid prop: type check failed for prop "noContainer". Expected Boolean, got String with value "true".

found in

---> <QMarkdown>
       <QPage>
         <CreatePost> at src/pages/post/Create.vue
           <QPageContainer>
             <QLayout>
               <MyLayout> at src/layouts/MyLayout.vue
                 <App> at src/App.vue
                   <Root>
`

**Comment by SouthRibbleTech** [comment]
I can also confirm that `:no-container="true"' works. Thanks @apasov I have spent 2 hours thinking I was going senile.

**Comment by SouthRibbleTech** [comment]
I am noticing that some markdown is not working i.e. ::: this is not displayed as a block :::

**Comment by hawkeye64** [comment]
@apasov @SouthRibbleTech Can you upgrade to v1.0.13 and tell me if you are still getting the container issue?

**Comment by Lecraminos** [comment]
I can confirm the above issue and "fix" in a similar setting (Chrome 79 on Manjaro) and unfortunately the error persists even after upgrading to 1.0.13... Used to work in 1.0.11, but interestingly even a downgrade (including removal of lock-file and node_modules) does not solve the issue anymore... Might it be introduced by a dependency upgrade?

**Comment by apasov** [comment]
@SouthRibbleTech lol)
@hawkeye64 Yes, I had already upgraded to v1.0.13 when I was writing the previous comment. The issue appeared after upgrading from v1.0.11 to v1.0.12

**Comment by hawkeye64** [comment]
So, fixed with v1.0.13?

**Comment by apasov** [comment]
No, not fixed.

**Comment by hawkeye64** [comment]
@apasov @Lecraminos  @mesqueeb I would appreciate it if you guys would install the latest and verify this fix put this issue to bed. Thanks. :heart:

**Comment by apasov** [comment]
I confirm. Works without `:no-container="true"`
Thanks!

**Comment by hawkeye64** [comment]
Perfect!

**Comment by SouthRibbleTech** [comment]
@hawkeye64 I can also confirm that this is working again. Thank you.

**Comment by mesqueeb** [comment]
@hawkeye64 sadly it didn't work for me. : (
I'm still getting the error.

**Comment by hawkeye64** [comment]
Yes, it's back. The fix breaks SSR. To implement the fix yourselves, don't install the app-extension (unless you need *.md or *.vmd imports). Instead, install the `quasar-ui-qmarkdown` component by itself. Then you have to write a boot file, and this is different for normal.

```js
import Vue from 'vue'
import VuePlugin from 'quasar-ui-qmarkdown/src/index.js'
// normally the above line would be like this: import VuePlugin from 'quasar-ui-qmarkdown

Vue.use(VuePlugin)
```
The problem lies within how the markdown-it plugins are created and imported. I have tried two different transpilers with rollup with different outcomes. I may have to implement babel, but really didn't want to. Or, make copies of all the plugins (maintenance nightmare) and write them properly for the transpiler.

**Comment by mesqueeb** [comment]
@hawkeye64 perhaps a great time to consider marked!
https://bundlephobia.com/result?p=marked@0.8.0

It's 4 times smaller and is more popular on Github. 😉

**Comment by hawkeye64** [comment]
@mesqueeb Thanks, I'll take a look at it

**Comment by mesqueeb** [comment]
@hawkeye64 it couldn't find the library when I did
```js
import QMarkdown from 'quasar-ui-qmarkdown/src/index.js'
```
But it did when I did this:
```js
import QMarkdown from '@quasar/quasar-ui-qmarkdown/src/index.js'
```

just a heads up for the others 😉

**Comment by hawkeye64** [comment]
@mesqueeb Thanks. I updated my example. Note, I found doing it that way breaks SSR.

Currently, investigating other markdown providers, including writing my own with AST parser to get better performance. This may take some time to come up with the best solution, but want something that is very long-term and not a band-aid (plaster).

**Comment by hawkeye64** [comment]
@apasov @Lecraminos @mesqueeb Such a tough bug. I have figured it out and have a fix. Look forward to a future release after I finish the rest of my tying.

**Comment by hawkeye64** [comment]
Fixed (finally) in v1.0.20
